### PR TITLE
 ILI-27: Added architecture for ORM_O01 parser with parsing MSH segment

### PR DESCRIPTION
**Pull request by oskarhinc**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisHL7Config.java [hunk 1]
```diff
@@ -47,6 +47,11 @@ public OrderIdentifier buildOrderIdentifier(Order order) {
 		return new OpenElisOrderIdentifier(order);
 	}
 	
+	@Override
+	public String getSendingFacilityNamespaceID() {
+		return null;
```
Comment (areklalo) [inline api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisHL7Config.java]: Consider using the UnsupportedOperationException or other.

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisOrderSender.java [hunk 2]
```diff
@@ -22,7 +22,7 @@
 	private OpenElisHL7Config config;
 	
 	@Autowired
-	private HL7OrderMessageGenerator msgGenerator;
+	private OMLO21OrderParser orderParser;
```
Comment (areklalo) [inline api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisOrderSender.java]: Like above

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/scc/SCCHL7Config.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+public class SCCHL7Config extends AbstractHL7Config {
+	
+	private static final String SENDING_FACILITY_NAMESPACE_ID = "LNSP";
+	
+	@Override
+	public String getReceivingApplication() {
+		return null;
```
Comment (areklalo) [inline api/src/main/java/org/openmrs/module/labintegration/api/hl7/scc/SCCHL7Config.java]: Do we have some default values?

File: api/src/test/java/org/openmrs/module/labintegration/api/communication/hl7/messages/OMLO21OrderParserTest.java [hunk 4]
```diff
@@ -16,30 +22,43 @@
 
 @ContextConfiguration(locations = { "classpath*:moduleApplicationContext.xml", "classpath*:applicationContext-service.xml",
         "classpath*:test-labContext.xml" }, inheritLocations = false)
-public class HL7OrderMessageGeneratorTest extends BaseModuleContextSensitiveTest {
+public class OMLO21OrderParserTest extends BaseModuleContextSensitiveTest {
```
Comment (areklalo) [inline api/src/test/java/org/openmrs/module/labintegration/api/communication/hl7/messages/OMLO21OrderParserTest.java]: Like above

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisHL7Config.java (hunk 1 above)
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisHL7Config.java]: It's not UnsupportedOperation in this case. For OpenElis messages this field is just not needed, so returning null here results in empty string in data in the message.

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/scc/SCCHL7Config.java (hunk 3 above)
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/labintegration/api/hl7/scc/SCCHL7Config.java]: It's just not-needed field in the message, like above

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisHL7Config.java (hunk 1 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisHL7Config.java]: Okay.

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/scc/SCCHL7Config.java (hunk 3 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/labintegration/api/hl7/scc/SCCHL7Config.java]: Okay.

File: api/src/test/java/org/openmrs/module/labintegration/api/communication/hl7/messages/OMLO21OrderParserTest.java (hunk 4 above)
Comment (oskarhinc) [reply api/src/test/java/org/openmrs/module/labintegration/api/communication/hl7/messages/OMLO21OrderParserTest.java]: Okay.

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisOrderSender.java (hunk 2 above)
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/labintegration/api/hl7/openelis/OpenElisOrderSender.java]: Okay.

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/messages/ORMO01OrderParser.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import org.openmrs.module.labintegration.api.hl7.config.HL7Config;
+import org.openmrs.module.labintegration.api.hl7.config.LabIntegrationProperties;
+import org.openmrs.module.labintegration.api.hl7.messages.gnerators.MshGenerator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Component
+public class ORMO01OrderParser implements OrderParser {
```
[the remark this answers is not in the document]
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/labintegration/api/hl7/messages/ORMO01OrderParser.java]: Okay.

File: api/src/main/java/org/openmrs/module/labintegration/api/hl7/messages/OMLO21OrderParser.java [hunk 6]
```diff
@@ -18,7 +18,7 @@
 import java.io.IOException;
 
 @Component
-public class HL7OrderMessageGenerator implements OrderParser {
```
[the remark this answers is not in the document]
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/labintegration/api/hl7/messages/OMLO21OrderParser.java]: Okay.

**Comment by oskarhinc** [comment]
I'll change the package name in another PR, to avoid conflicts with @pstudzinski

**Pull request merged**